timelock-agent/src/main/java/com/palantir/timelock/paxos/TimeLockAgent.java

```diff
@@ -16,85 +16,109 @@
 
 import io.reactivex.Observable;
 
-public class PaxosAgent extends TimeLockAgent {
-    private final PaxosInstallConfiguration paxosInstall;
+public class TimeLockAgent {
+    private final TimeLockInstallConfiguration install;
+    private final Observable<TimeLockRuntimeConfiguration> runtime;
+    private final Consumer<Object> registrar;
 
     private final PaxosResource paxosResource;
     private final PaxosLeadershipCreator leadershipCreator;
     private final LockCreator lockCreator;
     private final PaxosTimestampCreator timestampCreator;
     private final TimeLockServicesCreator timelockCreator;
 
-    public PaxosAgent(TimeLockInstallConfiguration install,
+    public TimeLockAgent(TimeLockInstallConfiguration install,
             Observable<TimeLockRuntimeConfiguration> runtime,
             Consumer<Object> registrar) {
         this(install, runtime, ImmutableTimeLockDeprecatedConfiguration.builder().build(), registrar);
     }
 
-    public PaxosAgent(TimeLockInstallConfiguration install,
+    public TimeLockAgent(TimeLockInstallConfiguration install,
             Observable<TimeLockRuntimeConfiguration> runtime,
             TimeLockDeprecatedConfiguration deprecated,
             Consumer<Object> registrar) {
-        super(install, runtime, deprecated, registrar);
-        this.paxosInstall = install.algorithm();
+        this.install = install;
+        this.runtime = runtime;
+        this.registrar = registrar;
 
-        this.paxosResource = PaxosResource.create(paxosInstall.dataDirectory().toString());
+        this.paxosResource = PaxosResource.create(install.paxos().dataDirectory().toString());
         this.leadershipCreator = new PaxosLeadershipCreator(install, runtime, registrar);
         this.lockCreator = new LockCreator(runtime, deprecated);
         this.timestampCreator = new PaxosTimestampCreator(paxosResource,
                 PaxosRemotingUtils.getRemoteServerPaths(install),
                 PaxosRemotingUtils.getSslConfigurationOptional(install).map(SslSocketFactories::createSslSocketFactory),
-                runtime.map(PaxosAgent::getAlgorithmConfigOrDefault));
+                runtime.map(TimeLockAgent::getAlgorithmConfigOrDefault));
         this.timelockCreator = install.asyncLock().useAsyncLockService()
                 ? new AsyncTimeLockServicesCreator(leadershipCreator, install.asyncLock())
                 : new LegacyTimeLockServicesCreator(leadershipCreator);
     }
 
-    @Override
     public void createAndRegisterResources() {
         registerPaxosResource();
-        registerPaxosExceptionMappers();
+        registerExceptionMappers();
         leadershipCreator.registerLeaderElectionService();
 
         // Finally, register the endpoints associated with the clients.
-        super.createAndRegisterResources();
+        registerTimeLockResourcesForClients();
+
+        ClockSkewMonitorCreator.create(install, registrar).registerClockServices();
```

Comment:
Is a separate conversation, but I'd be curious in understanding how this works? I assume it spins of some background thread that checks for clock skew? Might want to embed a way into the Agent to allow for wiring this up to your web servers executor service or w/e